js/src/tests/wrapped_openai.int.test.ts

```diff
@@ -574,3 +576,135 @@ test.concurrent("beta.chat.completions.parse", async () => {
+        },
+      ],
+    },
+    expectUsageMetadata: true,
+    checkReasoningTokens: true,
+  },
+];
+
+describe("Usage Metadata Tests", () => {
+  usageMetadataTestCases.forEach(
+    ({ description, params, expectUsageMetadata, checkReasoningTokens }) => {
+      it(`should handle ${description}`, async () => {
+        const { client, callSpy } = mockClient();
+        const openai = wrapOpenAI(new OpenAI(), {
+          tracingEnabled: true,
+          client,
+        });
+
+        const requestParams = { ...params };
+
+        let oaiUsage: OpenAI.CompletionUsage | undefined;
+        if (requestParams.stream) {
+          const stream = await openai.chat.completions.create(
+            requestParams as OpenAI.ChatCompletionCreateParamsStreaming
+          );
+          for await (const chunk of stream) {
+            if (expectUsageMetadata && chunk.usage) {
+              oaiUsage = chunk.usage;
+            }
+          }
+        } else {
+          const res = await openai.chat.completions.create(
+            requestParams as OpenAI.ChatCompletionCreateParams
+          );
+          oaiUsage = (res as OpenAI.ChatCompletion).usage;
+        }
+
+        let usageMetadata: UsageMetadata | undefined;
+        const requestBodies: any = {};
+        for (const call of callSpy.mock.calls) {
+          const request = call[2] as any;
+          const requestBody = JSON.parse(request.body);
+          if (request.method === "POST") {
+            requestBodies["post"] = [requestBody];
+          }
+          if (request.method === "PATCH") {
+            requestBodies["patch"] = [requestBody];
+          }
+          if (requestBody.outputs && requestBody.outputs.usage_metadata) {
+            usageMetadata = requestBody.outputs.usage_metadata;
+            break;
+          }
+        }
+
+        if (expectUsageMetadata) {
+          expect(usageMetadata).not.toBeUndefined();
+          expect(usageMetadata).not.toBeNull();
+          expect(oaiUsage).not.toBeUndefined();
+          expect(oaiUsage).not.toBeNull();
+          expect(usageMetadata!.input_tokens).toEqual(oaiUsage!.prompt_tokens);
+          expect(usageMetadata!.output_tokens).toEqual(
+            oaiUsage!.completion_tokens
+          );
+          expect(usageMetadata!.total_tokens).toEqual(oaiUsage!.total_tokens);
+
+          if (checkReasoningTokens) {
+            expect(usageMetadata!.output_token_details).not.toBeUndefined();
+            expect(
+              usageMetadata!.output_token_details!.reasoning
+            ).not.toBeUndefined();
+            expect(usageMetadata!.output_token_details!.reasoning).toEqual(
+              oaiUsage!.completion_tokens_details?.reasoning_tokens
+            );
+          }
+        } else {
+          expect(usageMetadata).toBeUndefined();
+          expect(oaiUsage).toBeUndefined();
+        }
+
+        if (process.env.WRITE_TOKEN_COUNTING_TEST_DATA === "1") {
```

Comment:
_nit_: might be nice to use `test.each` & `toMatchSnapshot`? Then we can use `jest --update-snapshot` instead of setting env var